modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/OpenApiServlet.java

```diff
@@ -80,4 +106,43 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         }
 
     }
+
+    public static Map<String, List<String>> getQueryParams(String queryString) {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        if (StringUtils.isEmpty(queryString)) {
+            return queryParameters;
+        }
+
+        String[] parameters = queryString.split("&");
+
+        for (String parameter : parameters) {
+            String[] keyValuePair = parameter.split("=");
+            queryParameters.put(keyValuePair[0], Collections.singletonList(keyValuePair[1]));
+        }
```

Comment:
This would overwrite query parameters which occur multiple times.

For example with `?param1=value1&param2=value2&param1=value3`, the `queryParameters` map would only contain `{param1=value3, param2=value2}`.

Consider using [`MultivaluedHashMap`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MultivaluedHashMap.html) instead.

Reply:
Good point. Fixed.